.github/workflows/linux.yml

```diff
@@ -0,0 +1,161 @@
+name: Linux
+
+on:
+  workflow_call:
+
+env:
+  IMAGE_NAME: ghcr.io/${{ github.repository }}
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: ./scripts/install_zig.sh
+      - run: PATH="zig/:$PATH" zig build lint
+    env:
+      SOURCE_REPO: ${{ github.event.pull_request.head.repo.clone_url }}
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        distro: [ubuntu, alpine]
+    steps:
+      - uses: actions/checkout@v3
+      - run: ./scripts/tests_on_${{ matrix.distro }}.sh
```

Comment:
This will probably run faster if you don't use the matrix and you list them out. No need to get multiple Github level jobs since these tests run in docker?

Reply:
The Github matrix jobs run on independent runners, so it ends up parallelizing them for us for free!